datafusion/core/src/datasource/data_source.rs

```diff
@@ -67,4 +69,33 @@ pub trait FileSource: Send + Sync {
     /// If this returns true, the DataSourceExec may repartition the data
     /// by breaking up the input files into multiple smaller groups.
     fn supports_repartition(&self, config: &FileScanConfig) -> bool;
```

Comment:
It seems like we could remove `supports_repartition` as well:
1. This API hasn't yet been released so it wouldn't be a breaking change
2. It would ensure that all places in the code that need to check repartition use a single API (and thus are consistently done)

Reply:
yeah absolutely, give me a second to pull it up.

Reply:
Add the PR here: https://github.com/apache/datafusion/issues/14123